lib/api/controllers/authController.js

```diff
@@ -91,4 +109,37 @@ module.exports = function AuthController (kuzzle) {
         return q.reject(new ResponseObject(modifiedRequestObject || requestObject, invalid));
       });
   };
+
+  /**
+   * Updates the current user if it is not anonymous
+   *
+   * @param {RequestObject} requestObject
+   * @param context
+   * @returns {Promise}
+   */
+  this.selfUpdate = function (requestObject, context) {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('security:beforeSelfUpdate', requestObject)
+      .then(newRequestObject => {
+        modifiedRequestObject = newRequestObject;
+        if (!context.token._id) {
+          return q.reject(new UnauthorizedError('User must be connected in order to call selfUpdate'));
+        }
+
+        if (modifiedRequestObject.data.body.profile) {
+          return q.reject(new BadRequestError('profile can not be provided in selfUpdate'));
+        }
+
+        if (modifiedRequestObject.data.body._id) {
+          return q.reject(new BadRequestError('_id can not be part of the body'));
+        }
+
+        return kuzzle.repositories.user.load(context.token.user._id);
+      })
+      .then(user => kuzzle.repositories.user.persist(_.extend(user, modifiedRequestObject.data.body), { database: { method: 'update' } }))
```

Comment:
Be careful with using `_.extend`: if user already contain a field it will not be overwritten by the request.
You have to use `_.merge` instead.

By this way, we need a functional test to test it : 
* Update user with a new field `foo: bar` 
* Then update user with an existing field `foo: baz`